google-cloud-examples/src/main/java/com/google/cloud/examples/compute/v1/ComputeInstanceExample.java

```diff
@@ -0,0 +1,288 @@
+    Compute.Instances.Insert request =
+        computeService.instances().insert(projectId, zone, requestBody);
+    Operation response = request.execute();
+    waitForOperationCompletion(response);
+    System.out.printf("Instance created: %s%n", instanceId);
+  }
+
+  /**
+   * Prints the list of instances contained within the specified zone.
+   *
+   * @throws IOException
+   */
+  public void printInstances() throws IOException {
+    System.out.println("Listing Compute Engine Instances");
+    Compute.Instances.List request = computeService.instances().list(projectId, zone);
+    InstanceList response;
+    do {
+      response = request.execute();
+      if (response.getItems() == null) {
+        continue;
+      }
+      for (Instance instance : response.getItems()) {
+        System.out.println(instance.toPrettyString());
+      }
+      request.setPageToken(response.getNextPageToken());
+    } while (response.getNextPageToken() != null);
+  }
+
+  /**
+   * Returns the specified Instance resource. Gets a list of available instances by making a list()
+   * request.
+   *
+   * @return an {@link Instance} object
+   * @throws IOException
+   */
+  public Instance getInstance() throws IOException {
+    System.out.println("Getting instance");
+    Compute.Instances.Get request = computeService.instances().get(projectId, zone, instanceId);
+    Instance response = request.execute();
+    return response;
+  }
+
+  /**
+   * Deletes the specified Instance resource.
+   *
+   * @throws Exception
+   */
+  public void deleteInstance() throws Exception {
+    System.out.println("Deleting an Instance");
+    Compute.Instances.Delete request =
+        computeService.instances().delete(projectId, zone, instanceId);
+    Operation response = request.execute();
+    waitForOperationCompletion(response);
+    System.out.printf("Deleted instance: %s%n", instanceId);
+  }
+
+  /**
+   * Retrieves aggregated list of all of the instances in your project across all regions and zones.
+   *
+   * @throws IOException
+   */
+  public void getAggregatedInstancesList() throws IOException {
+    System.out.println("Getting an aggregated list of all instances");
+    Compute.Instances.AggregatedList request = computeService.instances().aggregatedList(projectId);
+    InstanceAggregatedList response;
+    do {
+      response = request.execute();
+      if (response.getItems() == null) {
+        continue;
+      }
+      for (Map.Entry<String, InstancesScopedList> item : response.getItems().entrySet()) {
+        System.out.println(item.getKey() + " " + item.getValue());
+      }
+      request.setPageToken(response.getNextPageToken());
+    } while (response.getNextPageToken() != null);
+  }
+
+  /**
+   * Stops a running instance, shutting it down cleanly, and allows you to restart the instance at a
+   * later time. Stopped instances do not incur VM usage charges while they are stopped. However,
```

Comment:
Stops a running instance by shutting it down cleanly, and allows you to restart at a later time. Stopped instances do not incur VM usage charges; however, resources that the VM is using, such as persistent disks and static IP addresses continue to be charged until they are deleted.